cmd/fs-v1.go

```diff
@@ -268,6 +214,81 @@ func (fs *FSObjects) StorageInfo(ctx context.Context) StorageInfo {
 	return storageInfo
 }
 
+func (fs *FSObjects) waitForLowActiveIO() {
+	for {
+		if atomic.LoadInt64(&fs.activeIOCount) >= fs.maxActiveIOCount {
+			time.Sleep(lowActiveIOWaitTick)
+			continue
+		}
+		break
+	}
+}
+
+// crawlAndGetDataUsageInfo returns data usage stats of the current FS deployment
+func (fs *FSObjects) crawlAndGetDataUsageInfo(ctx context.Context, endCh chan struct{}) DataUsageInfo {
+
+	var dataUsageInfoMu sync.Mutex
+	var dataUsageInfo = DataUsageInfo{
+		BucketsSizes:          make(map[string]uint64),
+		ObjectsSizesHistogram: make(map[string]uint64),
+	}
+
+	walkFn := func(origPath string, typ os.FileMode) error {
```

Comment:
Or we could honour `endCh` to gracefully abort the walk

Reply:
> Or we could honour `endCh` to gracefully abort the walk

Unfortunately no, that's the drawback of using fastWalk, we cannot quit, even the standard filepath.Walk doesn't allow for quitting in the middle of walking operations

Reply:
> Unfortunately no, that's the drawback of using fastWalk, we cannot quit, even the standard filepath.Walk doesn't allow for quitting in the middle of walking operations

Now, to quit as quick as possible, filepath.SkipDir is always returned in walkFn when there is a current shutdown